### PR TITLE
Add make targets to build/run site in container, with "live editing"

### PR DESCRIPTION
**Pull request by dlipovetsky**
## Description of changes being made
Adds two make targets:
1. `make docker-liveedit-image` builds the site in a container
2. `make docker-liveedit` runs the site in the container, with live rendering of changes to files in the `pages` directory.

We would like to make use of this functionality to "live edit" konvoy docs, e..g,
```
make --debug docker-liveedit \
    LIVEEDIT_PAGES_SRC_ABS_PATH=/path/to/konvoy/repo/docs/site \
    LIVEEDIT_PAGES_DST_REL_PATH=ksphere/konvoy/latest \
    'LIVEEDIT_RENDER_PATH_PATTERN=ksphere/**'
```

The `docker-site-run` make target does run the site in a container, but requires docker-compose, and does not support live rendering.

This change does not affect any docs, or the existing toolchain.

## Checklist
- [x] Create your PR against `staging`, not `master`. 
- [x] Provide an estimated date for deploying the doc change. Note: Improvements or fixes can be merged ASAP. 

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/wiki/Contributing) for more information.

### CODE REVIEW

**Comment by dlipovetsky** [comment]
Thanks a lot for the review @samvantran!

> can you add docs/guide on how to deploy non-mesosphere pages?

Yes--I'll add this to the PR.

> f I refresh the page, I can see the change. Makes the reload server somewhat useless since it spends time to reload the files with no benefit.

Thanks for catching this. I recall noticing, but ignoring this behavior. I think I can fix it, though! The [live-reload mechanism](https://github.com/FWeinb/metalsmith-watch#livereload-default-false) a javascript client to query an HTTP server on a different port. I suspect that, if I publish this port, live reload will work :smile: .

**Comment by dlipovetsky** [comment]
Updated. The container now exposes the livereload server, and livereload works. Still need to add docs.

**Comment by dlipovetsky** [comment]
I amended my commit with an updated README. I'll also include your comment about rendering time.

**Comment by pierrebeitz** [comment]
@dlipovetsky being focussed on performance i forgot to mention that i would love to only provide a single env var in order to spin up the container. 

1. what do you think of using  `LIVEEDIT_PAGES_DST_REL_PATH` to automatically infer `LIVEEDIT_PAGES_SRC_ABS_PATH`?

2. what do you think of changing (or aliasing) `LIVEEDIT_PAGES_DST_REL_PATH` to something less technical to have an interface like e.g. `WATCH=ksphere/konvoy/ make docker-liveedit`?

**Comment by dlipovetsky** [comment]
> i noticed that the reload-speed seems to be much worse than in the bare metal setup and have two suggestions that can be found here: #3114. feel free to only rip one part or another from that or modify those changes however you like

Thanks a lot for the review @pierrebeitz. I know next to nothing about metalsmith, and appreciate your insight! If you're ok with merging this PR, I can follow up shortly on #3114

**Comment by dlipovetsky** [comment]
> i would love to only provide a single env var in order to spin up the container.

I like the idea.

> 1. what do you think of using `LIVEEDIT_PAGES_DST_REL_PATH` to automatically infer `LIVEEDIT_PAGES_SRC_ABS_PATH`?

`ABS_PATH` points to the path on the host, and `REL_PATH` points to the path in the container, relative to the repo root. For example:

- I'm updating konvoy docs in `/home/dlipovetsky/projects/konvoy/docs/site`
- I've checked out docs-docs-site in `/home/dlipovetsky/projects/dcos-docs-site`
- I'll run `make --debug docker-liveedit LIVEEDIT_PAGES_SRC_ABS_PATH=/home/dlipovetsky/projects/konvoy/docs/site LIVEEDIT_PAGES_DST_REL_PATH=ksphere/konvoy/latest`.

In this case, I'm not sure how I could infer `REL_PATH` from `ABS_PATH`.

> 2. what do you think of changing (or aliasing) `LIVEEDIT_PAGES_DST_REL_PATH` to something less technical to have an interface like e.g. `WATCH=ksphere/konvoy/ make docker-liveedit`?

That sounds good. And maybe `SOURCE` for `LIVEEDIT_PAGES_SRC_ABS_PATH`.

**Comment by pierrebeitz** [comment]
thanks for the feedback. i was not aware of the "cross-mounting" use case - thanks for pointing that out so digestible. 💡

> If you're ok with merging this PR, I can follow up shortly on #3114

:shipit: !

File: README.md [hunk 1]
```diff
@@ -117,14 +117,84 @@ This is to ensure setup was successful
...
-Running a build on first install is to ensure that all tools have been installed and are reading the code correctly. Please make sure to debug any errors at this point until you can get a full preview of the docs site at localhost:3000 in your browser. 
+Running a build on first install is to ensure that all tools have been installed and are reading the code correctly. Please make sure to debug any errors at this point until you can get a full preview of the docs site at localhost:3000 in your browser.
 
 :TODO: image localhost docs landing page
 
 ### Stop the local preview server
 This necessary once you are done previewing, not great to leave it running.
 `ctrl-c`
 
+## Deploy Live Site in a Container
```
Comment (cmastr) [inline README.md:128]: I didn't even know this file existed. We have been using the Wiki for this information. I need to do a cross reference or make sure this new info is in the wiki somewhere.

**Comment by cmastr** [comment]
Two more sources of truth:

https://hackmd.io/@ck4adventure/BJgrt4L5B/%2FxJo7K0hyR-GpmQNz7jU0Fw

https://wiki.d2iq.com/display/ENG/Contributing+to+the+dcos-docs-site+Github+Repository

**Pull request merged**